utils.go

```diff
@@ -0,0 +1,194 @@
+package main
+
+import (
+	"errors"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type check_with_timeout_type func() (bool, error)
+
+type ExecCommand struct {
+	Command    string
+	Directory  string
+	Env        []string
+	Input      []string
+	OutputFile string
+	Args       []string
+}
+
+var env_home string
+var env_singnet_repos string
+var env_go_path string
+var log_path string
+
+func init() {
+
+	env_home = os.Getenv("HOME")
+	env_singnet_repos = os.Getenv("SINGNET_REPOS")
+	env_go_path = os.Getenv("GOPATH")
+	log_path = env_go_path + "/log"
+	log.Printf("SINGNET_REPOS=%s\n", env_singnet_repos)
+}
+
+func read_file(file string) (string, error) {
+
+	buf, err := ioutil.ReadFile(file)
+	if err != nil {
+		return "", err
+	}
+	return string(buf), nil
+}
+
+func file_exists(file_name string) (bool, error) {
+
+	_, err := os.Stat(file_name)
+
+	if err == nil {
+		return true, nil
+	}
+
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+
+	return true, err
+}
+
+func write_to_file(file_name string, content string) error {
+
+	file, err := os.Create(file_name)
+	if err != nil {
+		return err
+	}
+
+	defer file.Close()
+
+	_, err = file.WriteString(content)
+	return err
+}
+
+func append_to_file(file_name string, content string) error {
+
+	file, err := os.OpenFile(file_name, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0600)
+	if err != nil {
+		return err
+	}
+
+	defer file.Close()
+
+	_, err = file.WriteString(content)
+	return err
+}
+
+func link_file(file_from string, file_to string) error {
```

Comment:
May be ```make_symlink```?

Reply:
Update to use os.Symlink instead of running ln from a command line.